### PR TITLE
[docs] update self-hosted db guides

### PR DESCRIPTION
**Pull request by GavinFrazar**
This docs PR updates redis cluster, cockroachdb, and mongodb self-hosted guides.

The commits are well ordered and it's probably best to review commit by commit to understand what motivated each change set.

For Redis cluster, it was brought up during the testplan that the setup steps in the `tls-cluster no` tab didn't work with the rest of the guide for creating the Redis cluster. I added that section in https://github.com/gravitational/teleport/pull/36260 - it sets redis cluster up such that Teleport's `db` CA is used to issues a cert for each node, yet still requires a custom CA to init the cluster + several additional steps. It's much simpler to just use the custom CA for each node's certs and only add trust for Teleport's `db_client` CA - if they want to use `tls-cluster no` they don't even need to change anything from the `tls-cluster yes` TLS setup.

For CockroachDB, I explained how to trust the Teleport `db_client` CA without changing the `node.crt` served by each node. Customers have asked about how they can do this and a few days ago it came up again. It's the default tab for that step because it's easier and simpler.
I also went into more detail about using `cockroach cert`.

For Mongo, I made some small improvements to indentation and only explain `tctl auth sign` setup for the config step that actually uses `tctl auth sign`.

### CODE REVIEW

**Review by greedy52 — COMMENTED** [review]
Thanks a lot for updating these!

**Review by greedy52 — APPROVED** [review]

File: docs/pages/database-access/guides/redis-cluster.mdx [hunk 1]
```diff
@@ -22,12 +22,17 @@ If you want to configure Redis Standalone, please read [Database Access with Red
 (!docs/pages/includes/edition-prereqs-tabs.mdx!)
 
 - Redis version `6.0` or newer.
+
+  <Admonition type="note" title="Note">
+    Redis `7.0` and RESP3 (REdis Serialization Protocol) are currently not supported.
```
[the remark this answers is not in the document]
Comment (GavinFrazar) [reply docs/pages/database-access/guides/redis-cluster.mdx:27]: ah ok, I'll update it. I just moved this note from below to be closer to the redis version pre-req info

File: docs/pages/database-access/guides/redis-cluster.mdx [hunk 2]
```diff
@@ -66,43 +71,16 @@ Install and configure Teleport where you will run the Teleport Database Service:
...
-using their hostnames and IP addresses.
-For example, if the first member is accessible at `redis1.example.com` with IP `10.0.0.1` and
-the second at `redis2.example.com` with IP `10.0.0.2`, run:
+Export your Teleport cluster's `db_client` CA cert and concatenate it with your Redis 
+Cluster's CA cert (in PEM format):
 ```code
-$ tctl auth sign --format=redis --host=redis1.example.com,10.0.0.1 --out=redis1 --ttl=2190h
-$ tctl auth sign --format=redis --host=redis2.example.com,10.0.0.2 --out=redis2 --ttl=2190h
+$ tctl auth export --type=db_client > db-client-ca.crt
+$ cat <Var name="/path/to/your/ca.crt" /> out.cas > pem-bundle.cas
```
[the remark this answers is not in the document]
Comment (GavinFrazar) [reply docs/pages/database-access/guides/redis-cluster.mdx:78]: thanks, after changing it to `tctl auth export` I should've made it reference `db-client-ca.crt`, not `out.cas`.

The `<Var... />` tag make it render as an editable var in the page so they can change it once and it updates here as well as in the teleport db definition below

File: docs/pages/includes/database-access/custom-db-ca.mdx [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+      protocol: "{{ protocol }}"
+      uri: "{{ scheme }}{{ protocol }}.example.com:{{ port }}{{ query }}"
+      static_labels:
+        "env": "example"
+      tls:
+        ca_cert_file: "<Var name='/path/to/your/ca.crt' />"
+  ```
+
+Now the Teleport Database Service will trust certificates presented by your
+{{ db }}.
```
[the remark this answers is not in the document]
Comment (GavinFrazar) [reply docs/pages/includes/database-access/custom-db-ca.mdx:15]: 459d9234e27efb625f993bbc693ac016a999e583

File: docs/pages/database-access/guides/redis-cluster.mdx [hunk 4]
```diff
@@ -167,18 +110,18 @@ in the Redis documentation for more details.
...
+$ export CERTS_DIR=/path/to/certs/
+$ export IP1=10.0.0.1 # update with the real node 1 IP
+$ export IP2=10.0.0.2 # update with the real node 2 IP
+$ export IP3=10.0.0.3 # update with the real node 3 IP
+$ export IP4=10.0.0.4 # update with the real node 4 IP
+$ export IP5=10.0.0.5 # update with the real node 5 IP
+$ export IP6=10.0.0.6 # update with the real node 6 IP
+$ redis-cli --user replica-user --cluster-replicas 1 --tls --cluster-yes \
+  --cluster create ${IP1}:7001 ${IP2}:7002 ${IP3}:7003 ${IP4}:7004 ${IP5}:7005 ${IP6}:7006 \
+  --cacert ${CERTS_DIR}/server.cas --key ${CERTS_DIR}/server.key --cert ${CERTS_DIR}/server.crt
```
[the remark this answers is not in the document]
Comment (GavinFrazar) [reply docs/pages/database-access/guides/redis-cluster.mdx]: explained what `servert.crt` is referring to, and offered the alternative to just create a new client cert: 13faf595f0cb771038592fbd413c083fa8b74de3

File: docs/pages/database-access/guides/cockroachdb-self-hosted.mdx [hunk 5]
```diff
@@ -81,49 +77,128 @@ in the CockroachDB documentation for more information.
...
+users can authenticate to your CockroachDB cluster as clients.
+
+The CockroachDB CA needs to be trusted by each CockroachDB node so that nodes 
+can authenticate themselves as clients to other nodes in the CockroachDB 
+cluster. This is because CockroachDB uses mTLS for node-to-node communication.
+
+<Tabs>
+<TabItem label="Using existing CockroachDB CA certs">
+In this configuration, your CockroachDB CA will be used to issue the server cert
+`node.crt` for each CockroachDB node.
```
[the remark this answers is not in the document]
Comment (GavinFrazar) [reply docs/pages/database-access/guides/cockroachdb-self-hosted.mdx]: Went with:
`"Nodes serving your CockroachDB CA certs"`
and
`"Nodes serving your Teleport CA certs"`

0a57c79dbe05b9b0b644fc494aa8a01e1283cb81

File: docs/pages/database-access/guides/redis-cluster.mdx (hunk 1 above)
[the remark this answers is not in the document]
Comment (GavinFrazar) [reply docs/pages/database-access/guides/redis-cluster.mdx]: cb082af0c5059befa41cad95f8e96c54911c9a74

**Review by ptgott — APPROVED** [review]

**Pull request merged**